### PR TITLE
feat(ui): add set email and phone passcode validation hooks

### PR DESCRIPTION
**Pull request by simeng-li**
## Summary

Add set email and phone passcode validation hooks to hand the continue flow's passwordless verification

## Testing

UT added

### CODE REVIEW

File: packages/ui/src/containers/PasscodeValidation/use-continue-set-email-passcode-validation.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+  const { run: setEmail } = useApi(continueWithEmail, setEmailErrorHandlers);
+
+  const onSubmit = useCallback(
+    async (code: string) => {
+      const verified = await verifyPasscode(email, code);
+
+      if (!verified) {
+        return;
+      }
```
Comment (charIeszhao) [inline packages/ui/src/containers/PasscodeValidation/use-continue-set-email-passcode-validation.ts:55]: nit: Would be better to add a test case to cover this branch

Comment (simeng-li) [reply packages/ui/src/containers/PasscodeValidation/use-continue-set-email-passcode-validation.ts:55]: it was covered by the passcode validation container test, in this PR as well.

File: packages/ui/src/apis/continue.ts [hunk 2]
```diff
@@ -82,3 +82,23 @@ export const sendContinueSetPhonePasscode = async (phone: string) => {
...
+
+export const verifyContinueSetSmsPasscode = async (phone: string, code: string) => {
+  await api
+    .post(`${passwordlessApiPrefix}/sms/verify`, {
+      json: { phone, code, flow: PasscodeType.Continue },
+    })
+    .json();
+
+  return { success: true };
+};
```
Comment (gao-sun) [inline packages/ui/src/apis/continue.ts:104]: should merge?

File: packages/ui/src/containers/PasscodeValidation/index.test.tsx [hunk 3]
```diff
@@ -234,36 +247,98 @@ describe('<PasscodeValidation />', () => {
         expect(mockedNavigate).toBeCalledWith('/forgot-password/reset', { replace: true });
       });
     });
+
+    it('fire Sms forgot-password validate passcode event', async () => {
```
Comment (gao-sun) [inline packages/ui/src/containers/PasscodeValidation/index.test.tsx:251]:
```suggestion
    it('fire sms forgot-password validate passcode event', async () => {
```

File: packages/ui/src/apis/continue.ts (hunk 2 above)
Comment (simeng-li) [reply packages/ui/src/apis/continue.ts:104]: Yes, that's a legacy API of code verification. Already have a separate issue refactoring that.

**Review by wangsijie — APPROVED** [review]

**Pull request merged**